packages/docs/docs/self-hosting/monitoring.md

```diff
@@ -0,0 +1,128 @@
+
+PostgreSQL metrics provide detailed insight into data layer performance.
+
+### Resource Usage
+**CPU Utilization** is critical, especially for the writer instance. Since there's only one writer (single point of failure), keep writer CPU below 60-70%. Reader instances can run hotter since you can scale horizontally by adding more readers.
+
+**Connection Count** tracks active database connections. PostgreSQL creates roughly one process per connection, so monitor to avoid resource exhaustion.
+
+### Query Efficiency
+**Tuples Read/Written** measure database activity at the row level. A "tuple" is essentially a row, though it can also exist in indexes.
+
+**Tuples Scanned vs. Fetched** reveals query efficiency. When scanned significantly exceeds fetched, the database is loading data it ultimately discards, indicating inefficient queries or missing indexes.
+
+**Transactions Per Second** shows overall database activity. The metric splits between commits (successful) and rollbacks (failed). Spikes in rollbacks usually indicate problems.
+
+## Cache Metrics (Redis)
+
+Redis serves as a fast in-memory cache to reduce database load.
+
+### Performance
+**CPU and Memory Usage** track Redis resource consumption. Memory usage directly correlates with stored data since Redis is an in-memory store.
+
+**Cache Hit/Miss Ratio** shows cache effectiveness. Some misses are normal, but unusual spikes might indicate application issues or invalid reference patterns.
+
+### Capacity Management
+**Evictions** occur when Redis runs out of memory and must remove least-recently-used items. Ideally this should be zero, but occasional evictions aren't necessarily problematic if your working set still fits in cache.
+
+**Alert Recommendation**: Monitor for evictions > 0 as an early warning of capacity issues.
+
+## Container Metrics
+
+If using containerized deployment (Docker, Kubernetes, ECS, etc.), these metrics track resource usage at the container level.
+
+### CPU Considerations for Node.js
+**CPU Utilization** requires special interpretation for Node.js applications. Since Node.js is single-threaded for application code, one fully-utilized core represents 100% application capacity.
+
+Calculate your threshold as: `1 / (number of CPU cores) * 100`
+
+For example:
+- 4-core system: Alert at 25% CPU
+- 8-core system: Alert at 12.5% CPU  
+- 32-core system: Alert at 3% CPU
+
+When CPU exceeds this threshold, the Node.js main thread is fully occupied and additional requests will queue up, causing exponential performance degradation.
+
+**Memory Usage** tracks container memory consumption, useful for detecting leaks or capacity planning.
+
+**Network I/O** provides supporting information about data transfer volumes, helpful for understanding traffic patterns and response sizes.
+
+## Alerting Strategy
+
+### High-Priority Alerts
+Set up immediate alerts for metrics that directly impact users:
+- Load balancer error rates (especially 504s)
+- Database writer CPU > 60-70%
```

Comment:
I'd suggest this be a bit lower of a range, maybe 50%-60%, but sustained for a few minutes. Thoughts, @codyebberson?

Reply:
That sounds good to me 👍